pkg/registration/spoke/spokeagent.go

```diff
@@ -418,6 +459,54 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 		)
 	}
 
+	var hubAcceptController, hubTimeoutController factory.Controller
+	hubKubeConfigSecretAutoGenCtx, cancelHubKubeConfgiSecretAutoGen := context.WithCancel(ctx)
+	defer cancelHubKubeConfgiSecretAutoGen()
```

Comment:
what is this for?

Reply:
Because we using checker and health probe to perform restart, we need to pervent the hubKubeConfigSecret to created by controller after we delete it.

The reason why we need to delete hub-kubeconfig is list up.

Reply:
Remove.